logging_log.go

```diff
@@ -0,0 +1,86 @@
+//go:build !go1.21
+
+package fauna
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+)
+
+type Logger interface {
```

Comment:
exposing two interfaces doesn't feel right to me. if our lib exposes a logging interface, shouldn't it be the same regardless of what version of go they're on?

Reply:
the result is a single interface. We allow consumers using Go 1.22+ to use the new `slog.Logger` hotness and 'fallback' to the older `log.Logger` for older those that are using older versions of Go. The build tags at the top of the files control this.